### DISCUSSION TITLE
UI slowing down significantly with increased #videos and predictions

### DISCUSSION THREAD

**Discussion by mfkeles** [talmolab/sleap#1058] [Help!]
Hi, 

I noticed that it is becoming impossible to use GUI for labeling when the project file contains many predictions.  I noticed that it significantly slowed down after I ran inference on a full video to see how it performed. After running predictions on 700k frames it became impossible to look at predictions and add new labels. Moving across different frames takes couple of seconds and menu bar slowed down significantly as well. Is there a way to improve this by just loading the annotated instances into the GUI and disregard predictions? Thank you in advance, 

Best,
Mehmet

**Comment by roomrys** [comment]
Hi @mfkeles,

### Root Cause
We recently made some changes to our cache which stores data for a more responsive GUI experience. What version of SLEAP are you running? And if running v1.2.9, did you experience this problem in previous versions? I wonder if this might be due to the change in our cache....

## Workaround
Unfortunately, SLEAP does not currently allow decoupling of loading just instances instead of predictions. As a general workflow, we suggest only predicting on the entire video once you are done with the human-in-the-loop training (i.e. predictions look good). During the human-in-the-loop training, we suggest using the labeling suggestions tab to predict on only a small subset of your project. 

If the cache is not working as expected, then slowdown will scale proportional to the number of labeled frames (predicted instances also count as labeled frames).

Thanks for reporting this and let me know if any of that helps in your case,
Liezl

**Comment by mfkeles** [comment]
Hi @roomrys 

Thank you for the explanation! I am using 1.2.9 and experiencing difficulties with the GUI. 

Is there a way to clear/delete the predictions without losing the instances? 

Thank you,
Mehmet

**Reply by roomrys** [reply]
Hi @mfkeles,

Oof, ok I'll need to investigate what's going on with our cache then. 

You can delete predictions through Labels > Delete Predictions. There is also the option to Custom Instance Delete which has more options for which instances/predictions to delete (and a few other Delete options) in the Labels menu.

Thanks,
Liezl
